### PR TITLE
[DX-1307, TT-11578] Set chart versions for bitnami database charts

### PR DESCRIPTION
**Pull request by komalsukhani**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1307](https://tyktech.atlassian.net/browse/DX-1307)

---

### Preview Link

<br>

1. [Tyk Stack Deployment - Postgres](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack-postgresql/)
2. [Tyk Stack Deployment - Mongo](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack-mongodb/)
3. [Tyk Stack Chart](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-stack-chart/)
4. [Tyk Control Plane Chart](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-control-plane-chart/)
5. [Tyk Data Plane Chart](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-data-plane-chart/#set-redis-connection-details-required)
6. [Tyk OSS Chart](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-oss-chart/#set-redis-connection-details-required)
7. [OTel](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_jaeger_k8s/#step-2-deploy-tyk-gateway-with-opentelemetry-enabled-using-helm)
8. [Prometheus Pump](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-pump/advanced-configurations/setup-prometheus-pump/)
9. [Tyk Cloud](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/tyk-cloud/environments-deployments/hybrid-gateways/#2-then-use-helm-to-install-redis-and-tyk)
10. [Tyk Cloud Legacy Helm](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/tyk-cloud/environments-deployments/hybrid-gateways-helm/)
11. [Tyk OSS Helm](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/tyk-oss/ce-helm-chart-new/)
12. [Tyk OSS Legacy Helm](https://deploy-preview-4439--tyk-docs.netlify.app/docs/nightly/tyk-oss/ce-helm-chart/)

### Description

<br>

Set version for bitnami charts.

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

___

## **Type**
enhancement, documentation

___

## **Description**
- Updated Redis, MongoDB, and PostgreSQL helm install commands across multiple documentation files to specify version numbers.
- Added notes about MongoDB image compatibility issues with MacOS M1 architecture in relevant sections.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>tyk-helm-tyk-stack-mongodb.md</strong><dd><code>Update Redis and MongoDB Helm Commands and Add Compatibility Note</code></dd></summary>
<hr>

tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack-mongodb.md
<li>Updated Redis and MongoDB helm install commands with specific version <br>numbers.<br> <li> Added a note about Bitnami MongoDB image compatibility with MacOS M1.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-0efbc1f980184fc652f595e445ac1c60f6ac3fd99e8aaa44aa68d58355e8f4a8">+8/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-helm-tyk-stack-postgresql.md</strong><dd><code>Update Redis and PostgreSQL Helm Commands</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack-postgresql.md
<li>Updated Redis and PostgreSQL helm install commands with specific <br>version numbers.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-0ada53f31b90e2cb4cee84193836845d1074be0009112bd8305337ff8c24ac90">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-control-plane-chart.md</strong><dd><code>Update Helm Commands and Add MongoDB Compatibility Note</code>&nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md
<li>Updated Redis and MongoDB helm install commands with specific version <br>numbers.<br> <li> Added a note about MongoDB compatibility with MacOS M1.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-1bb3ae7ce6b5aec344682558fe28a0e91ef5504473fbed884ae4651089784cf7">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-data-plane-chart.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-da211afa42cedf90ce80f9d59419356f09f0a76d37f56788b523bdf02b198e4f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-oss-chart.md</strong><dd><code>Update Helm Commands and Add MongoDB Compatibility Note</code>&nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md
<li>Updated Redis and MongoDB helm install commands with specific version <br>numbers.<br> <li> Added a note about MongoDB compatibility with MacOS M1.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-53fd570dd4fbaa9b39eca159ac973030b7f820da118b68df42df490432741dd5">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-stack-chart.md</strong><dd><code>Update Helm Commands and Add MongoDB Compatibility Note</code>&nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md
<li>Updated Redis and MongoDB helm install commands with specific version <br>numbers.<br> <li> Added a note about MongoDB compatibility with MacOS M1.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-f6e45bbf530fa763d587dec798584c8eb6294229ef353d3212ab01bf6b0a1d7c">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>otel_jaeger_k8s.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_jaeger_k8s.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-a10169e8eff56869b12268fc8c5e999f9ae7dba07c2b6752bb9e0a57a8dd3206">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>setup-prometheus-pump.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-pump/advanced-configurations/setup-prometheus-pump.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-ca42cfd654675ae2f1f1c567e8f898ce0ef960621facaa9306f275ca31aa5277">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>hybrid-gateways-helm.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways-helm.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-339fdf1636182483553106f59f021378e9f1dc9217b44677a43802714a74b8ac">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>hybrid-gateways.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-4edf7de9e6c13e231f8d8324c855a6a8a32b10a8ca6d4d38f031a9b70003013f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ce-helm-chart-new.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-oss/ce-helm-chart-new.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-31ebf23c056ca7f318b56a59ecf19f1ad64ace22ca45a562bfc5ed032b434ebf">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ce-helm-chart.md</strong><dd><code>Update Redis Helm Command</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-oss/ce-helm-chart.md
- Updated Redis helm install command with a specific version number.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-2c5e6b25589b0579e43c64b6747ba1bd8c8b2524697628c53c5a7f89776138dc">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-helm-chart.md</strong><dd><code>Update Helm Commands and Add MongoDB Compatibility Note</code>&nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-self-managed/tyk-helm-chart.md
<li>Updated Redis, MongoDB, and PostgreSQL helm install commands with <br>specific version numbers.<br> <li> Added a note about MongoDB compatibility with MacOS M1.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4439/files#diff-52a6ea0a5bb7d9c774ae3594dcf4fa0fe7b39ebe0b591f3e370820f0298feab0">+8/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

[DX-1307]: https://tyktech.atlassian.net/browse/DX-1307?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by olamilekan000 — APPROVED** [review]

**Review by dcs3spp — COMMENTED** [review]
Thanks for submitting @komalsukhani 

When @buraksekili and @caroltyk have peer reviewed I will merge and release.

**Comment by letzya** [comment]
@komalsukhani for future easier maintenance of this page, why don't we use env var like we use for namespace NAMESPACE?

**Comment by carolwkc** [comment]
@komalsukhani for redis installation, if we specify chart version, should we not set the Redis image tag? Especially we support Redis v7 from 5.3 too.

> helm upgrade tyk-redis oci://registry-1.docker.io/bitnamicharts/redis -n $NAMESPACE --create-namespace --install --set image.tag=6.2.13 --version 19.0.2

We might add a note that if user is on < Tyk 5.3, they need to check the Redis version supported by Tyk and use the corresponding Redis helm chart.

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack-mongodb.md [hunk 1]
```diff
@@ -70,9 +70,15 @@ The Bitnami chart also creates a secret `tyk-redis` which stores the connection
...
 
 ```bash
-helm upgrade tyk-mongo oci://registry-1.docker.io/bitnamicharts/mongodb -n $NAMESPACE --install
+helm upgrade tyk-mongo oci://registry-1.docker.io/bitnamicharts/mongodb -n $NAMESPACE --install --version 15.1.2
 ```
 
+{{< note >}}
+**Note**
+
+Bitnami MongoDB image is not supported on MacOS M1 architecture.
```
Comment (buraksekili) [inline tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack-mongodb.md]: Maybe we should say something like the following:
```suggestion
Bitnami MongoDB image is not supported on darwin/arm64 architecture.
```
What do you think?

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 2]
```diff
@@ -181,14 +181,20 @@ If not, you can use these rather excellent charts provided by Bitnami to install
...
-helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk --version 15.1.2
 ```
 
 Then follow notes from the installation output to get connection details and update them in `values.yaml` file.
 
 {{< note success >}}
 **Note**
 
+Bitnami MongoDB image is not supported on MacOS M1 architecture.
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: Same comment as above

File: tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md [hunk 3]
```diff
@@ -358,9 +358,15 @@ If you are using the MongoDB pumps in the tyk-oss installation you will require
...
 
 ```bash
-helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk --version 15.1.2
 ```
 
+{{< note success >}}
+**Note**
+
+Bitnami MongoDB image is not supported on MacOS M1 architecture.
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md]: Same comment as above

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 4]
```diff
@@ -181,14 +181,20 @@ If not, you can use these rather excellent charts provided by Bitnami to install
 **Mongo Installation**
 
 ```bash
-helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk --version 15.1.2
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: --version is specified twice

**Comment by komalsukhani** [comment]
> @komalsukhani for redis installation, if we specify chart version, should we not set the Redis image tag? Especially we support Redis v7 from 5.3 too.
> 
> > helm upgrade tyk-redis oci://registry-1.docker.io/bitnamicharts/redis -n $NAMESPACE --create-namespace --install --set image.tag=6.2.13 --version 19.0.2
> 
> We might add a note that if user is on < Tyk 5.3, they need to check the Redis version supported by Tyk and use the corresponding Redis helm chart.

@caroltyk I think we should make note more generic and add it below all the bitnami chart reference. 
Wdyt?

File: tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md [hunk 5]
```diff
@@ -358,9 +358,23 @@ If you are using the MongoDB pumps in the tyk-oss installation you will require
 To install MongoDB you can use these rather excellent charts provided by Bitnami:
 
 ```bash
-helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk --version 15.1.2
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md]: duplicated --version flag

File: tyk-docs/content/tyk-self-managed/tyk-helm-chart.md [hunk 6]
```diff
@@ -101,9 +107,14 @@ Alternatively, you can use `--set` flag to set it in Tyk installation. For examp
 <br />
 #### MongoDB
 ```bash
-helm install tyk-mongo bitnami/mongodb  --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+helm install tyk-mongo 
+/mongodb --set "replicaSet.enabled=true" -n tyk --version 15.1.2
```
Comment (carolwkc) [inline tyk-docs/content/tyk-self-managed/tyk-helm-chart.md]: do you mean `bitnami/mongodb`?

File: tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md [hunk 7]
```diff
@@ -175,9 +175,15 @@ If not, you can use these rather excellent charts provided by Bitnami to install
 **Mongo Installation**
 
 ```bash
-helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk
+helm install tyk-mongo bitnami/mongodb --version {HELM_CHART_VERSION} --set "replicaSet.enabled=true" -n tyk --version 15.1.2
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md]: duplicated --version flag

**Review by buraksekili — APPROVED** [review]

**Comment by dcs3spp** [comment]
@caroltyk Is the PR ok to merge now?

**Comment by carolwkc** [comment]
@dcs3spp Ok to merge. Thanks all!

**Pull request merged**